.github/workflows/ci.yml

```diff
@@ -7,43 +7,96 @@ on:
     branches: [master]
 
 jobs:
-  build:
-    strategy:
-      matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest ]
-    runs-on: ${{ matrix.os }}
+  initialize-ubuntu:
+    runs-on: ubuntu-latest
+    env:
+      SCHEME: scheme
     steps:
-      - uses: actions/checkout@v2
-
-      - uses: purescript-contrib/setup-purescript@main
-        with:
-          purescript: "0.15.8"
-          purs-tidy: "0.9.2"
-          psa: "0.8.2"
-          spago: "0.20.9"
-
-      - name: Cache PureScript dependencies
-        uses: actions/cache@v2
-        with:
-          key: ${{ runner.os }}-spago-${{ hashFiles('**/*.dhall') }}
-          path: |
-            .spago
-            output
-
-      - name: Set up Node toolchain
-        uses: actions/setup-node@v2
-        with:
-          node-version: "16"
+      - name: Install chez scheme
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y chezscheme
+      
+      - name: Verify Chez Scheme installed
+        run: |
+          ${SCHEME} --version
 
-      - name: Build the project
+  initialize-macos:
+    runs-on: macos-latest
+    env:
+      SCHEME: chez
+    steps:
+      - name: Install chez scheme
+        run: |
+          brew install chezscheme
+      
+      - name: Verify Chez Scheme installed
         run: |
-          spago build --purs-args "--censor-lib --strict"
+          ${SCHEME} --version
 
-      - name: Run tests 
+  initialize-windows:
+    runs-on: windows-latest
+    env:
+      MSYSTEM: MINGW64
+      MSYS2_PATH_TYPE: inherit
+      SCHEME: scheme
+    steps:
+      - name: Init
         run: |
-          spago -x test.dhall test
+          git config --global core.autocrlf false
 
-      - name: Check Formatting
-        if: runner.os == 'Linux'
+      - name: Get Chez Scheme
+        run: |
+          git clone --depth 1 https://github.com/cisco/ChezScheme
+          c:\msys64\usr\bin\bash -l -c "pacman -S --noconfirm tar make mingw-w64-x86_64-gcc"
+          echo "PWD=$(c:\msys64\usr\bin\cygpath -u $(pwd))" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+      - name: Configure and Build Chez Scheme
+        run: |
+          c:\msys64\usr\bin\bash -l -c "cd $env:PWD && cd ChezScheme && ./configure --threads && make"
+      - name: Set Path
         run: |
-          purs-tidy check src test
+          $chez="$(pwd)\ChezScheme\ta6nt\bin\ta6nt"
+          echo "$chez" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+      - name: Test Scheme
+        run: |
+          Invoke-Expression "$env:SCHEME --version"
```

Comment:
I don't know if I like this approach because I'd guess any future appearance of double quotes (e.g. for a CLI arg) will be brittle. However, due to my unfamiliarity with Windows, I'm not sure how else to invoke the env var.